rust/sample_connector_scaffolding/src/main.rs

```diff
@@ -0,0 +1,699 @@
+
+        // TODO: Verify the device endpoint specification is OK
+
+        // For this example, we will assume that the device endpoint specification is OK, see below for how to handle a bad update.
+        break local_device_endpoint_specification;
+
+        // // If the device specification is not OK await for a new one
+        // let _ = device_endpoint_status_reporter_watcher_rx.changed().await;
+        // // Update the status reporter for the new device endpoint specification
+        // device_endpoint_status_reporter_tx = device_endpoint_status_reporter_watcher_rx
+        //     .borrow_and_update()
+        //     .clone();
+    };
+
+    // Extract the asset status reporter from the asset watcher
+    let mut _asset_status_reporter_tx =
+        asset_status_reporter_watcher_rx.borrow_and_update().clone();
+
+    #[allow(clippy::never_loop)]
+    let mut _local_asset_specification = loop {
+        let local_asset_specification = dataset_client.asset_specification();
+
+        // TODO: Verify the asset specification is OK
+
+        // For this example, we will assume that the asset specification is OK, see below for how to handle a bad update.
+        break local_asset_specification;
+
+        // // If the asset specification is not OK await for a new one
+        // let _ = asset_status_reporter_watcher_rx.changed().await;
+        // // Update the status reporter for the new asset specification
+        // asset_status_reporter_tx = asset_status_reporter_watcher_rx.borrow_and_update().clone();
+    };
+
+    // This boolean tracks if the dataset and asset are ready to be sampled.
+    let mut is_dataset_asset_ready = true;
+    // This boolean tracks if the status for the dataset has been reported.
+    let mut is_dataset_reported = false;
+    // Extract the dataset definition from the dataset client
+    #[allow(clippy::never_loop)]
+    let mut local_dataset_definition = loop {
+        let local_dataset_definition = dataset_client.dataset_definition().clone();
+
+        // TODO: Verify the dataset definition is OK
+
+        // For this example, we will assume that the dataset definition is OK, see below for how to handle a bad update.
+        break local_dataset_definition;
+
+        // // If the dataset definition is not OK await for a new one
+        // match dataset_client.recv_notification().await {
+        //     DatasetNotification::Updated => {
+        //         log::info!("{dataset_log_identifier} Dataset definition updated");
+        //     }
+        //     DatasetNotification::Deleted => {
+        //         log::info!("{dataset_log_identifier} Dataset deleted, ending dataset handler");
+        //         return;
+        //     }
+        //     DatasetNotification::UpdatedInvalid => {
+        //         log::info!("{dataset_log_identifier} Dataset definition invalid, waiting for a valid update");
+        //         loop {
```

Comment:
can the parent match statement just be a loop instead of having the loop with the same match again inside?